crates/blockchain/blockchain.rs

```diff
@@ -331,30 +333,18 @@ impl Blockchain {
     pub async fn store_block(
         &self,
         block: &Block,
+        apply_updates_list: AccountUpdatesList,
         execution_result: BlockExecutionResult,
-        account_updates: &[AccountUpdate],
     ) -> Result<(), ChainError> {
-        // Apply the account updates over the last block's state and compute the new state root
-        let apply_updates_list = self
-            .storage
-            .apply_account_updates_batch(block.header.parent_hash, account_updates)
-            .await?
-            .ok_or(ChainError::ParentStateNotFound)?;
-
-        let new_state_root = apply_updates_list.state_trie_hash;
-        let state_updates = apply_updates_list.state_updates;
-        let accounts_updates = apply_updates_list.storage_updates;
-        let code_updates = apply_updates_list.code_updates;
-
         // Check state root matches the one in block header
-        validate_state_root(&block.header, new_state_root)?;
+        validate_state_root(&block.header, apply_updates_list.state_trie_hash)?;
```

Comment:
it seems this validation should be done _outside_ of this function? I would think that `store_block` is a function that you call after all validations were done, and you just need to call de db now (I know it's not this PRs fault but maybe we can change it here)